hyperspy/_components/polynomial.py

```diff
@@ -64,8 +64,11 @@ def grad_coefficients(self, x):
             return x * 0
 
     def __repr__(self):
-        return u'Component <%s order polynomial>'.encode('utf8') % (
-            ordinal(self.get_polynomial_order()))
+        text = u"%s order Polynomial component" % ordinal(
+            self.get_polynomial_order())
+        if self.name:
+            text = u"%s (%s)" % (self.name, text)
+        return u"<%s>".encode('utf8') % text
```

Comment:
The unicode - just out of habit.

At some point we some problems with strings (I forget what exactly), and I thought we encoded all `__repr__` as utf8?